playbooks/generate-csrs.yml

```diff
@@ -0,0 +1,95 @@
+---
+# Generate new certificate signing requests and keys for all cluster certificates,
+# to be delivered to a CA to sign. Keys are left in the csr staging directory
+# on each host, and csr files are copied to localhost in playbooks/csr.tgz
+
+- import_playbook: init/main.yml
+
+# Create CSRs
+- name: Create master CSRs
+  hosts: oo_masters_to_config
+  tasks:
+  - import_role:
+      name: openshift_master_certificates
+      tasks_from: csr.yml
+
+- name: Create etcd CSRs
+  hosts: oo_etcd_to_config
+  tasks:
+  - import_role:
+      name: etcd
+      tasks_from: csr.yml
+
+- name: Create node CSRs
+  hosts: oo_nodes_to_config
+  tasks:
+  - import_role:
+      name: openshift_node_certificates
+      tasks_from: csr.yml
+
+- name: Create hosted CSRs
+  hosts: oo_first_master
+  tasks:
+  - import_role:
+      name: openshift_hosted
```

Comment:
openshift_hosted_certificates ?
what is "hosted" ?

Reply:
openshift-router and registry certs